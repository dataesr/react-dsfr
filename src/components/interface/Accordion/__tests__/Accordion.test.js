import { render, screen, fireEvent } from '@testing-library/react';
import { Accordion, AccordionGroup } from '..';

describe('<Accordion />', () => {
  it('should render accordion properly', () => {
    render(
      <AccordionGroup className="custom-class">
        <Accordion title="first line accordion">
          Anything I want
        </Accordion>
        <Accordion title="first line accordion">
          Anything I want
        </Accordion>
      </AccordionGroup>,
    );
    const group = screen.getByTestId('accordion-group');
    const items = screen.getAllByTestId('accordion');
    expect(items).toHaveLength(2);
    expect(group).toMatchSnapshot();
  });

  it('should toggle on click', () => {
    render(
      <AccordionGroup className="custom-class">
        <Accordion title="first line accordion">
          Anything I want
        </Accordion>
        <Accordion title="first line accordion">
          Anything I want
        </Accordion>
      </AccordionGroup>,
    );
    const item = screen.getAllByTestId('accordion')[0];
    expect(item.isExpanded).toBeFalsy();
    const button = screen.getAllByTestId('accordion-button')[0];
    const itemDiv = screen.getAllByTestId('accordion-div')[0];
    expect(itemDiv.className).toBe('rf-collapse');
    fireEvent.click(button);
    expect(itemDiv.className).toBe('rf-collapse rf-collapse--expanded');
    fireEvent.click(button);
    expect(itemDiv.className).toBe('rf-collapse');
  });
  it('should toggle on click another accordion', () => {
    render(
      <AccordionGroup className="custom-class">
        <Accordion title="first line accordion">
          Anything I want
        </Accordion>
        <Accordion title="first line accordion">
          Anything I want
        </Accordion>
      </AccordionGroup>,
    );
    const buttons = screen.getAllByTestId('accordion-button');
    const itemDivs = screen.getAllByTestId('accordion-div');
    fireEvent.click(buttons[0]);
    expect(itemDivs[0].className).toBe('rf-collapse rf-collapse--expanded');
    fireEvent.click(buttons[1]);
    expect(itemDivs[0].className).toBe('rf-collapse');
    expect(itemDivs[1].className).toBe('rf-collapse rf-collapse--expanded');
  });
});
