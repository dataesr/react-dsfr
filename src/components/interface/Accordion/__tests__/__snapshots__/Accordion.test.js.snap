// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render accordion properly 1`] = `
<div
  class="custom-class"
  data-testid="accordiongroup"
>
  <ul
    class="fr-accordions-group"
  >
    <li
      class=""
      data-testid="accordion"
    >
      <section
        class="fr-accordion"
      >
        <h3
          class="fr-accordion__title"
        >
          <button
            aria-controls="fr-accordion-0"
            aria-expanded="false"
            class="fr-accordion__btn"
            data-testid="accordion-button"
            id="button0"
            type="button"
          >
            first line accordion
          </button>
        </h3>
        <div
          class="fr-collapse"
          data-testid="accordion-div"
          id="fr-accordion-0"
          style="--collapse: -0px;"
        >
          Anything I want
        </div>
      </section>
    </li>
    <li
      class=""
      data-testid="accordion"
    >
      <section
        class="fr-accordion"
      >
        <h3
          class="fr-accordion__title"
        >
          <button
            aria-controls="fr-accordion-1"
            aria-expanded="false"
            class="fr-accordion__btn"
            data-testid="accordion-button"
            id="button1"
            type="button"
          >
            first line accordion
          </button>
        </h3>
        <div
          class="fr-collapse"
          data-testid="accordion-div"
          id="fr-accordion-1"
          style="--collapse: -0px;"
        >
          Anything I want
        </div>
      </section>
    </li>
  </ul>
</div>
`;
